ecosystem/l2tol2cdm-relayedmessage-return.md

```diff
@@ -0,0 +1,85 @@
+# Purpose
+
+The ability to read events with Superchain interop is incredibly powerful. We should make sure we maximize the potential of all the default events emmitted when interacting with the OP-Stack predeploys.
+
+## Goals
+
+- Faciliate cross chain async programming
+
+# Problem Statement + Context
+
+As we horizontally scale, contract development will look more asynchronous. A lot of cross chain messages are predominalty write-based.
+
+- Bridge Funds
+- Invoke calldata on via a remotely controlled proxy account
+
+Read-based cross chain messages are less common but will increasingly become so in an asynchronous world. Solidity has a `view` modifier specifically targeted for reading -- it would be natural to also want to also query this state onchain very quickly.
```

Comment:
For some usecases, its not safe to read remote data without knowing the timestamp of the read. You could be tricked into reading stale data for some usecases. Is there an easy way to see the timestamp of the read?

Reply:
`Identifier` of the remote log plays in nicely here. The consumer on the source chain can choose to reject the `RelayedMessage` event if the timestamp of the event is stale

Nothing needs to change on the destination.